src/streamlink/plugins/nrk.py

```diff
@@ -1,64 +1,88 @@
+    _psapi_url = 'https://psapi.nrk.no'
+    # Program type to manifest type mapping
+    _program_type_map = {
+        'direkte': 'channel',
+        'serie': 'program',
+        'program': 'program',
+        'podkast': 'podcast',
+    }
 
-_mediaelement_schema = validate.Schema({
-    "mediaUrl": validate.url(
-        scheme="http",
-        path=validate.endswith(".m3u8")
-    )
-})
+    _url_re = re.compile(r"https?://(?:tv|radio).nrk.no/(program|direkte|serie|podkast)(?:/.+)?/([^/]+)")
 
+    _playable_schema = validate.Schema({
+        "playable": validate.any(
+            None,
+            {
+                "assets": [{
+                    "url": validate.url(
+                        path=validate.any(
+                            validate.endswith(".m3u8"),
+                            validate.endswith(".mp3")
+                        ),
+                    ),
+                    "format": str,
+                }],
+            },
+        ),
+        "nonPlayable": validate.any(
+            None,
+            {
+                "reason": str,
+            },
+        ),
+        "statistics": {
+            validate.optional("luna"): validate.any(
+                None,
+                {
+                    "data": {
+                        "title": str,
+                    },
+                },
+            ),
+        },
+    })
 
-class NRK(Plugin):
     @classmethod
     def can_handle_url(self, url):
-        return _url_re.match(url)
+        return self._url_re.match(url)
 
     def _get_streams(self):
-        # Get the stream type from the url (tv/radio).
-        stream_type = _url_re.match(self.url).group(1).upper()
-        cookie = {
-            "NRK_PLAYER_SETTINGS_{0}".format(stream_type): COOKIE_PARAMS
-        }
-
-        # Construct API URL for this program.
-        baseurl = self.session.http.get(self.url, cookies=cookie, schema=_schema)
-        program_id = _id_re.search(self.url).group(1)
+        # Construct manifest URL for this program.
+        program_type, program_id = self._url_re.search(self.url).groups()
+        manifest_type = self._program_type_map.get(program_type)
+        if manifest_type is None:
+            raise PluginError(f"Unknown program type '{program_type}'")
+        manifest_url = urljoin(self._psapi_url, f"playback/manifest/{manifest_type}/{program_id}")
 
         # Extract media URL.
-        json_url = urljoin(baseurl, "mediaelement/{0}".format(program_id))
-        res = self.session.http.get(json_url, cookies=cookie)
-        media_element = self.session.http.json(res, schema=_mediaelement_schema)
-        media_url = media_element["mediaUrl"]
+        res = self.session.http.get(manifest_url)
+        manifest = self.session.http.json(res, schema=self._playable_schema)
+        playable = manifest['playable']
+        if playable is None:
+            reason = manifest["nonPlayable"]["reason"]
+            raise PluginError(f"Not playable ({reason})")
```

Comment:
I changed both `PluginError` instances, but on second thought maybe the unknown `program_type` is considered a plugin error?

Reply:
Technically it should be a PluginError here, but since the regex already covers the possible values of the map and the error should thus never be raised, I don't think it's important. If you want to change it, you can do so, but if you don't then it's fine too.